test/dynamo/test_higher_order_ops.py

```diff
@@ -1037,7 +1037,12 @@ def false_fn(x):
 
         mod_for_compile = torch.compile(Foo(), backend=cnt, dynamic=True)
         mod_for_eager = Foo()
-        ref = mod_for_eager(torch.ones(6, 4))
+
+        with self.assertRaisesRegex(
+            torch._dynamo.exc.UserError,
+            r"Can't inplace modify module params/buffers inside HigherOrderOp",
+        ):
```

Comment:
As I mentioned in the last PR: having our own custom error and message would help surface to the user that "cond needs torch.compile to work, otherwise, it will fail"

Reply:
Yeah, I can do it here.